### ISSUE TITLE
Unexpected S3 reconciliation after a minor upgrade is causing high outgoing bandwidth usage, resource exhaustion

### ISSUE THREAD

**Issue by artiommocrenco** [ClickHouse/ClickHouse#65672]
**Describe the unexpected behaviour**
After upgrading clickhouse server to 24.3.4.147 from 24.3.2.23, I am observing increased use of outgoing bandwidth (close to maximum - 1Gbit/s) for all clickhouse replicas (total of 2 replicas). The traffic is directed towards S3 provider.

The unpredictable nature of these operations made me to downgrade clickhouse back to 24.3.2.23, and now  I am instead observing increased use of download bandwidth (the problem has reversed).

**Expected behavior**
Minimum reconciliation necessary after a minor/patch upgrade like this.

**Error message and/or stacktrace**
After enabling trace logging, I can observe many operations similar to the ones below:
```
REDACTED {} <Debug> IDisk: Copying from nvme2 (path: /mnt/nvme2/clickhouse/) REDACTED to clickhouse_s3_disk REDACTED
```

**Additional context**
No config changes were made.
Changelog for the versions affected: https://github.com/ClickHouse/ClickHouse/compare/v24.3.2.23-lts...v24.3.4.147-lts

**Comment by den-crane** [comment]
probably because of 
v24.3.4.147-lts.md:* Backported in [#64831](https://github.com/ClickHouse/ClickHouse/issues/64831): Fix bug which could lead to non-working TTLs with expressions. Fixes [#63700](https://github.com/ClickHouse/ClickHouse/issues/63700). [#64694](https://github.com/ClickHouse/ClickHouse/pull/64694) ([alesapin](https://github.com/alesapin)).

you just need to wait till all old partitions are moved to s3

you can monitor the process in system.moves / system.replication_queue

**Comment by artiommocrenco** [comment]
@den-crane thank you so much, this makes a lot of sense
